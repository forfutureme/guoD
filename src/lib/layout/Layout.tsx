/**
 * @format
 * @Author: huweijian
 * @Date: 2020-02-16 08:48:22
 * @Desc: Layout 布局组件
 */
import React from 'react'
import Aside from './Aside'
import './layout.scss'
import {joinClassName, prefixClass} from '@utils/classNameHandle'
const fixClass = prefixClass('guo-layout')

interface props extends React.HTMLAttributes<HTMLElement> {}

const Layout: React.FC<props> = ({className, children, ...rest}) => {
  const childrenArray = children as Array<React.ReactElement>
  const hasAside =
    childrenArray &&
    childrenArray.reduce &&
    childrenArray.reduce((result, node) => result || node.type === Aside, false)
  return (
    <div
      className={joinClassName(
        fixClass(),
        className,
        hasAside ? fixClass('has-aside') : ''
      )}
      {...rest}>
      {children}
    </div>
  )
}

export default Layout
export {default as Aside} from './Aside'
export {default as Content} from './Content'
export {default as Header} from './Header'
export {default as Footer} from './Footer'
