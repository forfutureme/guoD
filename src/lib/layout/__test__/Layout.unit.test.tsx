/**
 * @format
 * @Author: huweijian
 * @Date: 2020-02-16 10:10:18
 * @Desc: Layout单元测试
 */
import renderer from 'react-test-renderer'
import React from 'react'
import Layout, {Aside, Content, Header, Footer} from '../Layout'

describe('Layout测试', () => {
  it('测试Layout渲染', () => {
    const json = renderer
      .create(
        <Layout>
          <Aside></Aside>
          <Layout>
            <Header>Header</Header>
          </Layout>
        </Layout>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })
  it('测试Aside渲染', () => {
    const json = renderer.create(<Aside>Aside</Aside>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('测试Content渲染', () => {
    const json = renderer.create(<Content>Content</Content>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('测试Header渲染', () => {
    const json = renderer.create(<Header>Header</Header>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('测试Footer渲染', () => {
    const json = renderer.create(<Footer>Footer</Footer>).toJSON()
    expect(json).toMatchSnapshot()
  })
})
