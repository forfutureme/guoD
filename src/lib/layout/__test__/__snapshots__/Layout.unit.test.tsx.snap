// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout测试 测试Aside渲染 1`] = `
<div
  className="guo-layout-aside"
>
  Aside
</div>
`;

exports[`Layout测试 测试Content渲染 1`] = `
<div
  className="guo-layout-content"
>
  Content
</div>
`;

exports[`Layout测试 测试Footer渲染 1`] = `
<div
  className="guo-layout-footer"
>
  Footer
</div>
`;

exports[`Layout测试 测试Header渲染 1`] = `
<div
  className="guo-layout-header"
>
  Header
</div>
`;

exports[`Layout测试 测试Layout渲染 1`] = `
<div
  className="guo-layout guo-layout-has-aside"
>
  <div
    className="guo-layout-aside"
  />
  <div
    className="guo-layout"
  >
    <div
      className="guo-layout-header"
    >
      Header
    </div>
  </div>
</div>
`;
